ibm/service/uko/resource_ibm_key_template.go

```diff
@@ -0,0 +1,469 @@
+		return diag.FromErr(fmt.Errorf("Error setting key: %s", err))
+	}
+	keystores := []map[string]interface{}{}
+	for _, keystoresItem := range template.Keystores {
+		keystoresItemMap, err := ResourceIbmKeyTemplateKeystoresPropertiesToMap(&keystoresItem)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		keystores = append(keystores, keystoresItemMap)
+	}
+	if err = d.Set("keystores", keystores); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting keystores: %s", err))
+	}
+	if err = d.Set("description", template.Description); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting description: %s", err))
+	}
+	if err = d.Set("created_at", flex.DateTimeToString(template.CreatedAt)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting created_at: %s", err))
+	}
+	if err = d.Set("updated_at", flex.DateTimeToString(template.UpdatedAt)); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting updated_at: %s", err))
+	}
+	if err = d.Set("created_by", template.CreatedBy); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting created_by: %s", err))
+	}
+	if err = d.Set("updated_by", template.UpdatedBy); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting updated_by: %s", err))
+	}
+	if err = d.Set("href", template.Href); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting href: %s", err))
+	}
+	if err = d.Set("version", response.Headers.Get("Etag")); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting version: %s", err))
+	}
+
+	return nil
+}
+
+func ResourceIbmKeyTemplateUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	ukoClient, err := meta.(conns.ClientSession).UkoV4()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	updateKeyTemplateOptions := &ukov4.UpdateKeyTemplateOptions{}
+
+	id := strings.Split(d.Id(), "/")
+	region := id[0]
+	instance_id := id[1]
+	vault_id := id[2]
+	template_id := id[3]
+	updateKeyTemplateOptions.SetID(template_id)
+	updateKeyTemplateOptions.SetUKOVault(vault_id)
+
+	url, err := getUkoUrl(context, region, instance_id, ukoClient)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	ukoClient.SetServiceURL(url)
+
+	hasChange := false
+
+	if d.HasChange("key") || d.HasChange("keystores") {
+		keyprops, err := ResourceIbmKeyTemplateMapToKeyProperties(d.Get("key.0").(map[string]interface{}))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		var key *ukov4.KeyPropertiesUpdate
+		key.Size = keyprops.Size
+		key.ActivationDate = keyprops.ActivationDate
+		key.ExpirationDate = keyprops.ExpirationDate
+		key.State = keyprops.State
+		updateKeyTemplateOptions.SetKey(key)
+		// TODO: handle Keystores of type TypeList -- not primitive, not model
+		hasChange = true
+	}
+	if d.HasChange("description") {
+		updateKeyTemplateOptions.SetDescription(d.Get("description").(string))
+		hasChange = true
+	}
+
+	// Etag support
+	updateKeyTemplateOptions.SetIfMatch(d.Get("version").(string))
```

Comment:
Just think if we can make a Get call and retrive the latest version and use that version for Update (instead of d.Get("version").(string))